### PR TITLE
[LSP] Add missing project context handler for ts

### PR DESCRIPTION
**Pull request by dibarbet**
Fixes https://github.com/dotnet/roslyn/issues/63844

### CODE REVIEW

File: src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptProjectContextHandler.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using Microsoft.CodeAnalysis.LanguageServer.Handler.DocumentChanges;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript;
+
+[ExportStatelessLspService(typeof(GetTextDocumentWithContextHandler), ProtocolConstants.TypeScriptLanguageContract), Shared]
+internal class VSTypeScriptProjectContextHandler : GetTextDocumentWithContextHandler
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public VSTypeScriptProjectContextHandler()
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptProjectContextHandler.cs:19]: We don't actually need an implementation here?

Comment (dibarbet) [reply src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptProjectContextHandler.cs:19]: The base implementation works for TS, so we don't need any custom behavior here.

Comment (jasonmalinowski) [reply src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptProjectContextHandler.cs:19]: @dibarbet Does any language ever need to customize it? I'm not sure why this isn't just an automatic for everything.

Comment (dibarbet) [reply src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptProjectContextHandler.cs:19]: Any roslyn based one shouldn't need to.  But some handlers are different so the MEF exports are per server are needed.  I did try and add support for 'export this for all server' but it ended up being more complicated than it was worth when having to deal with restarts, etc.

**Review by jasonmalinowski — APPROVED** [review]

**Pull request merged**